### DISCUSSION TITLE
How to add new column

### DISCUSSION THREAD

**Discussion by santoshbs** [h2oai/datatable#2702] [Q&A] [answered]
Is there a way to add a new column to a datatable that is based on a function of another column?

**Comment by st-pasha** [comment, marked as answer]
Yes, you can either use an assignment statement, or the `update()` function:
```
DT["newcol"] = <expression> # or
DT[:, update(newcol=<expression>)]
```
where `<expression>`  is any formula involving other columns. For example:
```
DT[:, update(Revenue = f.Quantity * f.Price)]

**Reply by santoshbs** [reply]
I am still failing to understand the datatable equivalent of the following pandas dataframe code:

`df['col2']= df.apply(lambda row: my_function(row['col1']), axis=1)`

**Reply by st-pasha** [reply]
Currently there is no direct equivalent of the pandas `apply` function, though the same effect can be achieved by converting  the column into a  python list, applying the function to the list, and then converting back  into a Frame:
```
df['col2'] = dt.Frame([my_function(x)  for x in df['col1'].to_list()[0]])
```
